### PR TITLE
REF: remove unused variable and checks

### PR DESCRIPTION
**Pull request by jbrockmendel**
- [ ] closes #xxxx (Replace xxxx with the GitHub issue number)
- [ ] [Tests added and passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#writing-tests) if fixing a bug or adding a new feature
- [ ] All [code checks passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit).
- [ ] Added [type annotations](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#type-hints) to new arguments/methods/functions.
- [ ] Added an entry in the latest `doc/source/whatsnew/vX.X.X.rst` file if fixing a bug or adding a new feature.

### CODE REVIEW

File: pandas/core/arrays/datetimes.py [hunk 1]
```diff
@@ -158,15 +158,8 @@ def f(self):
             # these return a boolean by-definition
             return result
 
-        if field in self._object_ops:
```
Comment (mroeschke) [inline pandas/core/arrays/datetimes.py:161]: Can `_object_ops` be in-lined in `_datetimelike_ops` now since this usage is removed?

Comment (jbrockmendel) [reply pandas/core/arrays/datetimes.py:161]: I think so. Will update.

**Comment by jorisvandenbossche** [comment]
@jbrockmendel should this be backported along with your previous PRs touching `_str_map`? I assume this one is not critical because just a small cleanup, but for avoiding conflicts if you make further changes to those methods, better to backport as well? (if so, then this is a small reminder to please include "string dtype" in your PR title)

**Comment by jbrockmendel** [comment]
This doesnt need backport, no.

**Review by mroeschke — APPROVED** [review]

**Comment by mroeschke** [comment]
Thanks @jbrockmendel

**Pull request merged**